hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -393,5 +396,28 @@ default boolean snapshottingEnabled() {
          * Returns the guarantee for current job.
          */
         ProcessingGuarantee processingGuarantee();
+
+
+        /**
+         * Returns the attached directory to the job with the given id.
+         *
+         * @param id identifier defined on the {@link ResourceConfig} to be
+         *           used retrieve files from storage.
+         * @return {@link File} handle to the attached directory to the job with provided identifier.
+         * @since 4.0
+         */
+        @Nonnull
+        File getAttachedDirectory(@Nonnull String id);
```

Comment:
you can skip the `get` (so just have `attachedDirectory` as it's the convention on this interface